### PR TITLE
Pangle: add appid & placementid to bidder param

### PR DESCRIPTION
**Pull request by chshpra**
doc PR: https://github.com/prebid/prebid.github.io/pull/2942

changes:
- add the optional appid and placementid (which when used, must be used in conjunction) to bidder param, then use them to form networkids in imp.ext
- add a true is_prebid flag for the auto recognition of prebid requests on our side

### CODE REVIEW

File: adapters/pangle/pangle.go
Comment (SyntaxNode) [inline adapters/pangle/pangle.go:27]: I'm curious, why does Prebid need to be flagged specially?

File: static/bidder-params/pangle.json
Comment (SyntaxNode) [inline static/bidder-params/pangle.json:15]: Would this be the same effect as an integer type? Do you expect appid and placementid to have leading zeros or exceed an int32's length?

File: adapters/pangle/pangle.go
Comment (VeronikaSolovei9) [inline adapters/pangle/pangle.go:103]: Question: do you expect to have one of AppId and PlacementId or both of them? In description you mentioned `add the optional appid and placementid (which when used, must be used in conjunction)`

File: adapters/pangle/pangle.go
Comment (chshpra) [reply adapters/pangle/pangle.go:27]: Our service handles Prebid requests and other ordinary OpenRTB requests differently. I'm hoping that we could automatically distinguish that by the request itself, not by some other manual configurations.

File: static/bidder-params/pangle.json
Comment (chshpra) [reply static/bidder-params/pangle.json:15]: Placementid could exceeds an int32's length, appid don't. they're both set to be of string type because of a historical problem on our side. We prefer to make this compatible with the existing codes.

File: adapters/pangle/pangle.go
Comment (chshpra) [reply adapters/pangle/pangle.go]: yeah this is a negligence. I have modified this part. Please take a look at it again. btw, isn't there a bidder param check before going into this adapter? can I assume that it's always both or none here at this part of codes?

Comment (SyntaxNode) [reply adapters/pangle/pangle.go]: >  btw, isn't there a bidder param check before going into this adapter? can I assume that it's always both or none here at this part of codes?

Yes. The bidder param check isn't part of the json tests, but is guaranteed to run by PBS-Core. You may assume the check passed in your bidder code.

**Review by bsardo — APPROVED** [review]
LGTM

**Comment by chshpra** [comment]
hi @bsardo, if there is no more problems, could we merge this pr now? our clients are waiting for this version of new adapter. thanks!

**Comment by bsardo** [comment]
Hi @chshpra. I'm just waiting for another reviewer to sign off on it; we require two approvals before merging. I'll try to get someone else to look at this today.

**Comment by chshpra** [comment]
@bsardo I see. thanks!

**Comment by VeronikaSolovei9** [comment]
LGTM

**Review by VeronikaSolovei9 — APPROVED** [review]

**Pull request merged**
